BEAMS3D/Sources/beams3d_physics_mod.f90

```diff
@@ -7,2058 +7,2099 @@
-           sigma = sqrt( ABS((one-zeta_o*zeta_o)*speed_cube) ) ! The standard deviation.
-           zeta_mean = zeta_o *(one - speed_cube )  ! The new mean in the distribution.
-           zeta = zeta*sigma + zeta_mean  ! The new pitch angle.
-           !!!The pitch angle MUST NOT go outside [-1,1] nor be NaN; but could happen accidentally with the distribution.
-           zeta = MIN(MAX(zeta,-0.999D+00),0.999D+00)
-           !IF (ABS(zeta) >  0.999D+00) zeta =  SIGN(0.999D+00,zeta)
-           vll = zeta*speed
-
-           !------------------------------------------------------------
-           !  Kick Model Scattering (old)
-           !------------------------------------------------------------
-           !IF (modb>=B_kick_min .and. modb<=B_kick_max) THEN
-           !   zeta_o = vll/speed   ! Record the current pitch.
-           !   zeta = zeta_o-zeta_o*(one-zeta_o*zeta_o)*dt*fact_kick*SQRT(ne_temp)/(modb*modb)
-           !   vll = zeta*speed
-           !END IF
-
-           !------------------------------------------------------------
-           !  Kick Model Scattering (new Energy, vll constant)
-           !------------------------------------------------------------
-           IF (modb>=B_kick_min .and. modb<=B_kick_max) THEN
-              zeta_o = vll/speed   ! Record the current pitch.
-              speed = speed*SQRT(one + fact_kick*modb*(1-zeta_o*zeta_o)*dt/SQRT(ne_temp))
-           END IF
-
-           !------------------------------------------------------------
-           !  Final Moment and vll update (return q(4))
-           !------------------------------------------------------------
-           moment = half*mymass*(speed*speed - vll*vll)/modb
-           q(4) = vll
-
+            q(4) = vll
+            RETURN
+         END IF
+         l = MAX(MIN(CEILING(SQRT(s_temp)*ns_prof1),ns_prof1),1)
+         epower_prof(mybeam,l) = epower_prof(mybeam,l) + mymass*dve*dt*speed*weight(myline)
+         ipower_prof(mybeam,l) = ipower_prof(mybeam,l) + mymass*dvi*dt*speed*weight(myline)
+         vll = vfrac*vll
+         moment = vfrac*vfrac*moment
+         speed = newspeed
+
+         ! !------------------------------------------------------------
+         ! !  Velocity diffusion - TODO: integrate into heating calculation
+         ! !------------------------------------------------------------
+         ! speed_cube = (speed*speed*speed)
+         ! CALL gauss_rand(1,zeta)  ! A random from a standard normal (1,1)
+         ! sigma = sqrt( ABS(2*e_charge*dt*(te_temp*speed_cube+ti_temp*vcrit_cube)*tau_spit_inv*inv_mymass/speed_cube) ) ! The standard deviation.
+         ! speed = speed+sigma*zeta
+
+
+         !------------------------------------------------------------
+         !  Pitch Angle Scattering
+         !------------------------------------------------------------
+         speed_cube = vc3_tauinv*slow_par(3)*dt/(speed*speed*speed) ! redefine as inverse
```

Comment:
????  This is too many changes to a fundamental part of the code.  The operators should not be changed with out a published benchmark of tests.